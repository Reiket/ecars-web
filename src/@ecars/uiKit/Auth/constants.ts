import {withBlockClass} from 'ecars-web-lib';
import {AuthPageWrapper} from '@ecars/uiKit/Auth/components/AuthPageWrapper';

export const AuthPageWrapperHOC = withBlockClass(AuthPageWrapper, 'auth');