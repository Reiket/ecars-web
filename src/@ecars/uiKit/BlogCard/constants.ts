import type {CategoriesListType} from 'ecars-web-lib';

export const BLOG_CARD_DIRECTION = Object.freeze({
  ROW: 'row',
  COLUMN: 'column',
});

export type BlogCardDirectionType = (typeof BLOG_CARD_DIRECTION)[keyof typeof BLOG_CARD_DIRECTION];

export interface BlogCardContentProps {
  title: string;
  category: CategoriesListType;
  description?: string;
}

export interface BlogCardImageProps {
  imageUrl: string;
  alt?: string;
}

export const BLOG_CARD_TEST_ID = 'blogCardTestId';
