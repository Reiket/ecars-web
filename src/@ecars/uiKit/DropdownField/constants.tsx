import type {DropdownOption} from 'ecars-web-lib';

interface TopbarDropdownsConfig {
  label: string;
  options: DropdownOption[];
}

const shipToConfig: DropdownOption[] = [
  {
    value: 'ca',
    name: 'Canada',
    label: (
      <img
        src="./assets/images/flags/canada.svg"
        alt="ca"
      />
    ),
  },
  {
    value: 'us',
    name: 'USA',
    label: (
      <img
        src="./assets/images/flags/usa.svg"
        alt="usa"
      />
    ),
  },
  {
    value: 'eu',
    name: 'Europe',
    label: (
      <img
        src="./assets/images/flags/europe.svg"
        alt="eu"
      />
    ),
  },
];

const currencyConfig: DropdownOption[] = [
  {
    value: 'usd',
    name: 'US Dollar',
  },
  {
    value: 'eur',
    name: 'Euro',
  },
  {
    value: 'gbp',
    name: 'British Pound',
  },
];

export const topbarDropdownsConfig: TopbarDropdownsConfig[] = [
  {
    label: 'Ship to:',
    options: shipToConfig,
  },
  {
    label: 'Currency:',
    options: currencyConfig,
  },
];
