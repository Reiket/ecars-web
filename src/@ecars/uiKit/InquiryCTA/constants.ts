import {withBlockClass} from 'ecars-web-lib';
import {SectionWrapper} from '@ecars/uiKit/Section/components/SectionWrapper';

export const InquiryCtaWrapperHOC = withBlockClass(SectionWrapper, 'inquiry-cta');
