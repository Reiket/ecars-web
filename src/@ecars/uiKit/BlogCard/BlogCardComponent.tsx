import type {FC} from 'react';
import {BlogCard} from '@ecars/uiKit/BlogCard/index';
import type {BlogCardContentProps, BlogCardDirectionType, BlogCardImageProps} from '@ecars/uiKit/BlogCard/constants';
import {BLOG_CARD_TEST_ID} from '@ecars/uiKit/BlogCard/constants';
import type {ElementProps} from 'ecars-web-lib';
import {cn} from 'ecars-web-lib';

interface Props extends BlogCardContentProps, BlogCardImageProps, ElementProps {
  direction: BlogCardDirectionType;
}

export const BlogCardComponent: FC<Props> = ({block, description, title, category, imageUrl, alt, direction}) => {
  const classNames = cn(block, 'blog-card', `blog-card--${direction}`);
  return (
    <article
      data-testid={BLOG_CARD_TEST_ID}
      className={classNames}
    >
      <BlogCard.Image
        imageUrl={imageUrl}
        alt={alt}
      />
      <BlogCard.Content
        category={category}
        title={title}
        description={description}
      />
    </article>
  );
};
