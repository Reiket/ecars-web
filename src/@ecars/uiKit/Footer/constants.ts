import {withBlockClass} from 'ecars-web-lib';
import {FooterWrapper} from '@ecars/uiKit/Footer/components/FooterWrapper';
import {socialLinks} from '@ecars/uiKit/SocialList/constants';
import {PageUrls} from '@ecars/constants/page-urls';
import {FooterCopyWrapper} from '@ecars/uiKit/Footer/components/FooterCopyWrapper';

export const FooterWrapperHOC = withBlockClass(FooterWrapper, 'footer');
export const FooterCopyWrapperHOC = withBlockClass(FooterCopyWrapper, 'copy');
export const footerSocialLinks = socialLinks.filter((item) => !('label' in item));
export const footerContactsLinks = socialLinks.filter((item) => 'label' in item);

export const copyLinks = [
  {
    label: 'Privacy Policy',
    href: PageUrls.PRIVACY,
  },
  {
    label: 'Terms & Conditions',
    href: PageUrls.TERMS,
  },
];

export interface FooterLink {
  label: string;
  to: string;
}

export const companyLinks: FooterLink[] = [
  {label: 'About Us', to: PageUrls.ABOUT},
  {label: 'Blog', to: PageUrls.Blog},
  {label: 'FAQ', to: PageUrls.FAQ},
];

export const carsLinks: FooterLink[] = [
  {label: 'Special offers', to: '/'},
  {label: 'New cars', to: PageUrls.NEWCARS},
  {label: 'Used cars', to: PageUrls.USEDCARS},
  {label: 'Brands', to: '/'},
];
