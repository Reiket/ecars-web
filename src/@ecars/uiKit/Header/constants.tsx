import {Icons, withBlockClass} from 'ecars-web-lib';
import {HeaderWrapper} from '@ecars/uiKit/Header/components/HeaderWrapper';
import type {PageUrlType} from '@ecars/constants/page-urls';
import {PageUrls} from '@ecars/constants/page-urls';
import {HeaderNavbarWrapper} from '@ecars/uiKit/Header/components/HeaderNavbarWrapper';
import type {ReactNode} from 'react';

export const HeaderWrapperHOC = withBlockClass(HeaderWrapper, 'header');
export const HeaderNavbarWrapperHOC = withBlockClass(HeaderNavbarWrapper, 'navbar');

interface NavbarLink {
  to: PageUrlType;
  label: string;
  isUtil?: boolean;
}

interface HeaderUtil {
  to: PageUrlType;
  icon: ReactNode;
}

export const navbarLinksConfig: NavbarLink[] = [
  {
    to: PageUrls.CATALOG,
    label: 'All Cars',
  },
  {
    to: PageUrls.ABOUT,
    label: 'About Us',
  },
  {
    to: PageUrls.Blog,
    label: 'Blog',
  },
  {
    to: PageUrls.FAQ,
    label: 'FAQ',
  },
  {
    to: PageUrls.FAVORITES,
    label: 'Favorites',
    isUtil: true,
  },
  {
    to: PageUrls.LOGIN,
    label: 'Login',
    isUtil: true,
  },
];

export const headerUtilsConfig: HeaderUtil[] = [
  {to: PageUrls.FAVORITES, icon: <Icons.FavoriteSolid />},
  {to: PageUrls.LOGIN, icon: <Icons.Account />},
];

export const headerTestScenarios = [
  {
    desc: 'initial state',
    scrolls: [0],
    expected: {sticky: false, showHeader: true},
  },
  {
    desc: 'scrolled down below topbarHeight → sticky true, showHeader false',
    scrolls: [100],
    expected: {sticky: true, showHeader: false},
  },
  {
    desc: 'scroll up again → sticky true, showHeader true',
    scrolls: [150, 50],
    expected: {sticky: true, showHeader: true},
  },
];
