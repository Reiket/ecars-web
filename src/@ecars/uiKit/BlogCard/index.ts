import {BlogCardContent} from '@ecars/uiKit/BlogCard/components/BlogCardContent';
import {BlogCardComponent} from '@ecars/uiKit/BlogCard/BlogCardComponent';
import {BlogCardImage} from '@ecars/uiKit/BlogCard/components/BlogCardImage';

export const BlogCard = Object.assign(BlogCardComponent, {
  Content: BlogCardContent,
  Image: BlogCardImage,
});
