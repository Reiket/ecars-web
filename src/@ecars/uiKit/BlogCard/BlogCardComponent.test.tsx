import {render, screen} from '@testing-library/react';
import {BLOG_CARD_DIRECTION, BLOG_CARD_TEST_ID} from '@ecars/uiKit/BlogCard/constants';
import {BlogCard} from '@ecars/uiKit/BlogCard';
import {capitalizeFirstLetter, CATEGORIES_LIST} from 'ecars-web-lib';
import {expect, describe, test} from 'vitest';

const defaultProps = {
  title: 'title',
  imageUrl: '',
  category: CATEGORIES_LIST.NEWS,
};

describe('Blog Card Component', () => {
  const cardDirection = Object.values(BLOG_CARD_DIRECTION);
  const descriptionOptions = ['Description', undefined];
  cardDirection.forEach((direction) => {
    descriptionOptions.forEach((option) => {
      const hasDescription = option ? 'has a description' : '';
      test(`renders correctly with ${direction} direction, category, title ${hasDescription}`, () => {
        const {container} = render(
          <BlogCard
            description={option}
            direction={direction}
            {...defaultProps}
          />,
        );
        const element = screen.getByTestId(BLOG_CARD_TEST_ID);
        expect(element).toBeInTheDocument();
        if (option) {
          expect(screen.getByText(option)).toBeInTheDocument();
        }
        expect(screen.getByText(defaultProps.title)).toBeInTheDocument();
        expect(screen.getByText(capitalizeFirstLetter(defaultProps.category))).toBeInTheDocument();
        expect(element).toHaveClass(`blog-card--${direction}`);
        expect(container).toMatchSnapshot();
      });
    });
  });
});
