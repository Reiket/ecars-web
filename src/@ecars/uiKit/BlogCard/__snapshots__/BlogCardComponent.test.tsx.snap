// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Card Component renders correctly with column direction, category, title 1`] = `
<div>
  <article
    class="blog-card blog-card--column"
    data-testid="blogCardTestId"
  >
    <img
      alt="Blog_Card_Image"
      class="blog-card__image"
      src=""
    />
    <div
      class="blog-card__content"
    >
      <span
        class="category category--news"
        data-testid="categoryTestId"
      >
        News
      </span>
      <h3
        class="blog-card__title title-blog-card"
      >
        title
      </h3>
    </div>
  </article>
</div>
`;

exports[`Blog Card Component renders correctly with row direction, category, title 1`] = `
<div>
  <article
    class="blog-card blog-card--row"
    data-testid="blogCardTestId"
  >
    <img
      alt="Blog_Card_Image"
      class="blog-card__image"
      src=""
    />
    <div
      class="blog-card__content"
    >
      <span
        class="category category--news"
        data-testid="categoryTestId"
      >
        News
      </span>
      <h3
        class="blog-card__title title-blog-card"
      >
        title
      </h3>
    </div>
  </article>
</div>
`;

exports[`Blog Card Component renders with description 1`] = `
<div>
  <article
    class="blog-card blog-card--row"
    data-testid="blogCardTestId"
  >
    <img
      alt="Blog_Card_Image"
      class="blog-card__image"
      src=""
    />
    <div
      class="blog-card__content"
    >
      <span
        class="category category--news"
        data-testid="categoryTestId"
      >
        News
      </span>
      <h3
        class="blog-card__title title-blog-card"
      >
        title
      </h3>
      <p
        class="blog-card__description"
      >
        Description
      </p>
    </div>
  </article>
</div>
`;

exports[`Blog Card Component renders with description 2`] = `
<div>
  <article
    class="blog-card blog-card--column"
    data-testid="blogCardTestId"
  >
    <img
      alt="Blog_Card_Image"
      class="blog-card__image"
      src=""
    />
    <div
      class="blog-card__content"
    >
      <span
        class="category category--news"
        data-testid="categoryTestId"
      >
        News
      </span>
      <h3
        class="blog-card__title title-blog-card"
      >
        title
      </h3>
      <p
        class="blog-card__description"
      >
        Description
      </p>
    </div>
  </article>
</div>
`;
